mona_openai/mona_openai_legacy.py

```diff
@@ -207,105 +155,8 @@ async def _inner_create(
-                )
-
-            log_message = add_conditional_sampling(
-                _inner_log_message, sampling_ratio
-            )
-
-            start_time = time.time()
-
-            async def inner_super_function():
-                # Call the actual openai create function without the Mona
-                # specific arguments.
-                return await call_non_blocking_sync_or_async(
-                    super_function,
-                    args,
-                    {
-                        x: kwargs[x]
-                        for x in kwargs
-                        if not x.startswith(MONA_ARGS_PREFIX)
-                    },
-                )
-
-            async def inner_handle_exception():
-                if not specs.get("avoid_monitoring_exceptions", False):
-                    await log_message(True)
-
-            if not is_stream:
-                try:
-                    response = await inner_super_function()
-                except Exception:
-                    await inner_handle_exception()
-                    raise
-
-                await log_message(False)
-
-                return response
-
-            # From here it's stream handling.
-
-            async def _stream_done_callback(
-                final_response, actual_stream_start_time
-            ):
-                nonlocal response
-                nonlocal stream_start_time
-                # There is no usage data in returned stream responses, so
-                # we add it here.
-                response = final_response | {
-                    "usage": get_usage(
-                        model=get_model_param(kwargs),
-                        prompt_texts=base_class._get_all_prompt_texts(kwargs),
-                        response_texts=base_class._get_all_response_texts(
-                            final_response
-                        ),
-                    )
-                }
-                stream_start_time = actual_stream_start_time
-                await log_message(False)
-
-            try:
-                # Call the actual openai create function without the Mona
-                # specific arguments.
-                return ResponseGatheringIterator(
-                    original_iterator=await inner_super_function(),
-                    delta_choice_text_getter=(
-                        base_class._get_stream_delta_text_from_choice
-                    ),
-                    final_choice_getter=base_class._get_final_choice,
-                    callback=_stream_done_callback,
-                )
-
-            except Exception:
-                await inner_handle_exception()
-                raise
+            return await create_logic(super_function.__name__ == "acreate", export_function,super_function,cls._get_logging_message,sampling_ratio,base_class._get_all_prompt_texts, base_class._get_all_response_texts,base_class._get_stream_delta_text_from_choice, base_class._get_final_choice, specs,args,kwargs)
```

Comment:
nit: spaces after commas

Reply:
also when we send such a list of arguments (above as well) it might help readability to add a comment where this list's spec comes from

Reply:
black for the nit. WDYM where it comes from? From the function's definition... What kind of comment are you imagining here?